CartApplication/CartItem.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace CartApplication
+{
+    public class CartItem
+    {
+        public Product product;
+        public int quantity;
+        public double totalPrice;
+
+        public CartItem(Product product, int quantity = 1)
+        {
+            this.product = product;
+            this.quantity = quantity;
+        }
+
+        public void CalculateCartItemTotalPrice()
```

Comment:
You should use public getter instead of method here..
Its bad practice to allow change internal state of object from outside